runelite-client/src/main/java/net/runelite/client/plugins/prayerindicator/PrayerIndicatorOverlay.java

```diff
@@ -0,0 +1,269 @@
+
+import com.google.inject.Inject;
+import java.awt.BasicStroke;
+import java.awt.Color;
+import java.awt.Dimension;
+import java.awt.Graphics2D;
+import java.awt.geom.Ellipse2D;
+import java.awt.geom.Rectangle2D;
+import java.awt.image.BufferedImage;
+import net.runelite.api.Client;
+import net.runelite.api.InventoryID;
+import net.runelite.api.Item;
+import net.runelite.api.ItemID;
+import net.runelite.api.Query;
+import net.runelite.api.Skill;
+import net.runelite.api.queries.InventoryItemQuery;
+import net.runelite.api.widgets.Widget;
+import net.runelite.api.widgets.WidgetInfo;
+import net.runelite.client.game.ItemManager;
+import net.runelite.client.plugins.itemstats.potions.PrayerPotion;
+import net.runelite.client.ui.overlay.Overlay;
+import net.runelite.client.ui.overlay.OverlayLayer;
+import net.runelite.client.ui.overlay.OverlayPosition;
+import net.runelite.client.ui.overlay.infobox.Counter;
+import net.runelite.client.ui.overlay.infobox.InfoBoxManager;
+import net.runelite.client.util.QueryRunner;
+
+public class PrayerIndicatorOverlay extends Overlay
+{
+	
+	//Set up some constants that will be used later on
+	private final float THICKNESS = 2f;			// This will set how thick the circle should be
+	
+	private Color orbIndicatorColor = new Color (0, 255, 255, 255);
+	private int alphaOffset = -10;			// How much to subtract/add to the alpha of the indicator orb
+	private int playerPotionID;				// The current ItemID of the potion the player is carrying
+	private BufferedImage prayPotIcon;
+	private Counter indicator;				// The indicator that will go in the info box if the player chooses to have it shown
+	
+	//private PrayerIndicatorPlugin plugin;
+	private final Client client;
+	private final PrayerIndicatorConfig config;
+	private final InfoBoxManager infoBoxManager;
+	private final QueryRunner queryRunner;
+	
+	@Inject
+	private ItemManager itemManager;
+	
+	@Inject
+	private PrayerIndicatorPlugin plugin;
+	
+	@Inject
+	public PrayerIndicatorOverlay (QueryRunner queryRunner, Client client, PrayerIndicatorConfig config, InfoBoxManager infoBoxManager)
+	{
+		setPosition (OverlayPosition.DYNAMIC);
+		setLayer (OverlayLayer.ABOVE_WIDGETS);
+		this.client = client;
+		this.config = config;
+		this.infoBoxManager = infoBoxManager;
+		this.queryRunner = queryRunner;
+	}
+	
+	// This method will be called every game tick and will be used to fade the orb indicator in and out
+	public void onTick ()
+	{
+		//Every game tick let's get what kind of prayer potion the player has
+		playerPotionID = getPlayerPotion ();
+		
+		//Let's make sure the infoBox indicator is set
+		if (indicator == null)
+		{
+			prayPotIcon = itemManager.getImage (ItemID.PRAYER_POTION4);
+			indicator = new Counter (prayPotIcon, plugin, "");
+		}
+	}
+	
+	// This method will go every time the config was changed and is only used for if the player deactivates the info box display so it can be properly be removed
+	public void onConfigChanged ()
```

Comment:
You should look at other plugins, but I think this should stay in the Plugin class and instead add `indicator` as a getter/setter variable that the Overlay can access.